sympy/series/limits.py

```diff
@@ -321,7 +322,10 @@ def set_signs(expr):
         if z0 is S.Infinity:
             if e.is_Mul:
                 e = factor_terms(e)
-            newe = e.subs(z, 1/z)
+            if isinstance(e, Piecewise):
+                newe = e._at_infinity(z)
+            else:
+                newe = e.subs(z, 1/z)
```

Comment:
Either the `_at_infinity` method should be added to all expressions or we should just add a function here rather than adding a method to `Piecewise`.

Reply:
It is better to start with a function for these things rather than adding methods. Unless there is a plant to implement this method in a nontrivial way for many different types or some reason for extensibility I would not use a method.